lib/retry.js

```diff
@@ -19,6 +19,10 @@ import constant from 'lodash/constant';
  * * `interval` - The time to wait between retries, in milliseconds.  The
  *   default is `0`. The interval may also be specified as a function of the
  *   retry count (see example).
+ * * `filter` - Synchronous function that is invoked on erroneous result with the
```

Comment:
I don't think filter is the right term for this. Perhaps `continueOperation` or something? /cc @hargasinski @aearly 
Also this should be documented as optional